dbt_copilot_helper/templates/pipelines/codebase/overrides/stack.ts

```diff
@@ -432,6 +454,10 @@ export class TransformedStack extends cdk.Stack {
         this.codestarConnection = {arn: codestarConnectionArn, id: codestarConnectionId};
     }
 
+    private loadEnvManagerRole() {
+        this.envManagerRole = `arn:aws:iam::${this.account}:role/${this.appName}-*-EnvManagerRole`;
```

Comment:
In fact if the policyDocument were extracted, this won't even be required.

Reply:
@antroy-madetech me and @WillGibson tried a deploy with just the role attached to the PolicyDocument but the build failed so *we think* it needs both for a successful run

Reply:
one thing that can be improved, however is to get rid of the BuildProject's assumeRole blocks for dev and staging as the wildcard we add is going to shadow them

Reply:
I think @ant is getting at...

We could extract the policy document to a method, inline the ``arn:aws:iam::${this.account}:role/${this.appName}-*-EnvManagerRole`` bit, then lose this method.

Reply:
@WillGibson @antroy-madetech sorted